PocketKit/Sources/PocketKit/ReaderSettings/ReaderSettings.swift

```diff
@@ -2,35 +2,196 @@
+        case .justified:
+            return .justified
+        }
+    }
+
+    var isUsingDefaults: Bool {
+        fontSizeAdjustment == 0 &&
+        fontFamily == .blanco &&
+        lineHeightScaleFactorIndex == 3 &&
+        marginsIndex == 3 &&
+        alignmentOption == .natural
+    }
+
+    var isPremium: Bool {
+        userStatus == .premium
+    }
+
+    var lineHeightScaleFactor: Double {
+        ReaderAppearance.lineHeightMultipliers[lineHeightScaleFactorIndex]
+    }
+
+    var margins: Double {
+        ReaderAppearance.margins(for: uiSetting)[marginsIndex]
+    }
 
     var currentStyling: FontStyling {
         if fontFamily == .graphik {
             return GraphikLCGStyling()
         } else {
-            return BlancoOSFStyling()
+            return GenericFontStyling(family: fontFamily)
         }
     }
 
-    init(userDefaults: UserDefaults) {
+    var fontSet: [FontDescriptor.Family] {
+        if isPremium {
+            return ReaderAppearance.freeFontFamilies + ReaderAppearance.premiumFontFamilies
+        }
+        return ReaderAppearance.freeFontFamilies
+    }
+
+    var fontSizeAdjustmentRange: ClosedRange<Int> {
+        -6...6
+    }
+
+    var fontSizeAdjustmentStep: Int {
+        2
+    }
+    /// Default range of indexes for settings adjustments
+    var settingIndexRange: ClosedRange<Int> {
+        0...6
+    }
+
+    /// Step for setting indexes
+    var settingIndexStep: Int {
+        1
+    }
+
+    init(tracker: Tracker, userDefaults: UserDefaults) {
         _fontSizeAdjustment = AppStorage(wrappedValue: 0, UserDefaults.Key.readerFontSizeAdjustment, store: userDefaults)
-        _fontFamily = AppStorage(wrappedValue: .blanco, UserDefaults.Key.readerFontSizeAdjustment, store: userDefaults)
+        _fontFamily = AppStorage(wrappedValue: .blanco, UserDefaults.Key.readerFontFamily, store: userDefaults)
+        _lineHeightScaleFactorIndex = AppStorage(wrappedValue: 3, UserDefaults.Key.readerScaleFactorIndex, store: userDefaults)
+        _marginsIndex = AppStorage(wrappedValue: 3, UserDefaults.Key.readerMarginsIndex, store: userDefaults)
+        _userStatus = AppStorage(wrappedValue: .unknown, UserDefaults.Key.userStatus, store: userDefaults)
+        _alignmentOption = AppStorage(wrappedValue: .natural, UserDefaults.Key.readerTextAlignment, store: userDefaults)
+        self.tracker = tracker
+    }
+
+    func reset() {
+        fontSizeAdjustment = 0
+        fontFamily = .blanco
+        lineHeightScaleFactorIndex = 3
+        marginsIndex = 3
+        alignmentOption = .natural
     }
 }
 
-extension FontDescriptor.Family: RawRepresentable {
-    public init?(rawValue: String) {
-        self = FontDescriptor.Family(name: rawValue)
+// MARK: Premium upsell
+extension ReaderSettings {
+    @MainActor
+    func makePremiumUpgradeViewModel() -> PremiumUpgradeViewModel {
+        PremiumUpgradeViewModel(
+            store: Services.shared.subscriptionStore,
```

Comment:
This is ugly, but our Singleton-based DI needs a refactor that's out of scope here, and it would take a lot of passthrough injection to have this injected, so, for now, we can leave it as it is and plan the refactor.